core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -54,23 +63,54 @@
 
     private final CompletableFuture<List<Endpoint>> initialEndpointsFuture = new EventLoopCheckingFuture<>();
     private final AsyncCloseableSupport closeable = AsyncCloseableSupport.of(this::closeAsync);
+    private boolean allowEmptyEndpoints;
 
     /**
-     * Creates a new empty {@link DynamicEndpointGroup} that uses
-     * {@link EndpointSelectionStrategy#weightedRoundRobin()} as its {@link EndpointSelectionStrategy}.
+     * Creates a new empty instance, using {@link EndpointSelectionStrategy#weightedRoundRobin()}
+     * and disallowing an empty {@link Endpoint} list.
      */
     public DynamicEndpointGroup() {
         this(EndpointSelectionStrategy.weightedRoundRobin());
     }
 
     /**
-     * Creates a new empty {@link DynamicEndpointGroup} that uses the specified
-     * {@link EndpointSelectionStrategy}.
+     * Creates a new empty instance, disallowing an empty {@link Endpoint} list.
+     *
+     * @param selectionStrategy the {@link EndpointSelectionStrategy} of this {@link EndpointGroup}
      */
     public DynamicEndpointGroup(EndpointSelectionStrategy selectionStrategy) {
         this.selectionStrategy = requireNonNull(selectionStrategy, "selectionStrategy");
     }
 
+    /**
+     * Creates a new empty instance, using {@link EndpointSelectionStrategy#weightedRoundRobin()}.
+     *
+     * @param allowEmptyEndpoints whether to allow an empty {@link Endpoint} list
+     */
+    protected DynamicEndpointGroup(boolean allowEmptyEndpoints) {
```

Comment:
nit; I'm not sure if there will be a need for this constructor. What do you/other maintainers think?

Reply:
Actually nevermind this comment. I guess we can leave since it is `protected`

Reply:
Yeah, I changed it to `protected` so that a subclass can still use it. :-)